source/common/stats/isolated_store_impl.h

```diff
@@ -25,20 +25,35 @@ namespace Stats {
 template <class Base> class IsolatedStatsCache {
 public:
   using Allocator = std::function<std::shared_ptr<Base>(StatName name)>;
+  using AllocatorImportMode = std::function<std::shared_ptr<Base>(StatName, Gauge::ImportMode)>;
 
   IsolatedStatsCache(Allocator alloc) : alloc_(alloc) {}
+  IsolatedStatsCache(AllocatorImportMode alloc) : alloc_import_(alloc) {}
 
   Base& get(StatName name) {
     auto stat = stats_.find(name);
     if (stat != stats_.end()) {
       return *stat->second;
     }
 
+    ASSERT(alloc_);
```

Comment:
nit: I don't think this assert is needed, it will crash obviously on the next line. Same below.

Reply:
removing (will be in next push).